app/src/main/java/com/github/bgabriel998/softwaredevproject/CameraUiView.java

```diff
@@ -363,10 +372,25 @@ private void drawMountainMarker(POIPoint poiPoint, Boolean isVisible, int actual
         float mountainMarkerPosition = height * (rangeDegreesVertical - 2*deltaVerticalAngle) / (2*rangeDegreesVertical)
                 - (float)mountainMarkerVisible.getHeight()/2;
 
+        //Calculate the horizontal position
+        float left = pixDeg * (actualDegree - minDegrees);
+
         //Draw the marker on the preview depending on the line of sight
         Bitmap mountainMarker = isVisible ? mountainMarkerVisible : mountainMarkerNotVisible;
-        canvas.drawBitmap(mountainMarker, pixDeg * (actualDegree - minDegrees),
-                mountainMarkerPosition, null);
+        mountainInfo.setColor(isVisible ? getResources().getColor(mountainInfoColor, null) : compassColor);
+        mountainInfo.setAlpha(MAX_ALPHA);
+
+        canvas.drawBitmap(mountainMarker, left, mountainMarkerPosition, null);
+
+        //Save status before Screen Rotation
+        canvas.save();
+        canvas.rotate(LABEL_ROTATION, left, mountainMarkerPosition);
+        canvas.drawText(poiPoint.getName() + " " + poiPoint.getAltitude() + "m",
+                left + mountainInfo.getTextSize() - OFFSET_MOUNTAIN_INFO,
+                mountainMarkerPosition + mountainInfo.getTextSize() + OFFSET_MOUNTAIN_INFO,
+                mountainInfo);
```

Comment:
It looks good. I just wondered if it is possible to create some kind of tests for this to see if the correct test is set?

Reply:
Well the text is the name of the POIPoint so to test if the text is correct, we need to test the POIPoint.getName() method which is probably already done.